nimbleparse/src/diagnostics.rs

```diff
@@ -153,11 +153,14 @@ impl<'a> SpannedDiagnosticFormatter<'a> {
             } else {
                 out.push('\n');
                 let s = match e.spanskind() {
-                    SpansKind::DuplicationError => {
+                    SpansKind::DuplicationError | SpansKind::OptionalDuplicationSpan => {
```

Comment:
This is a bit awkward, we can only count the duplicate occurances by their spans, so for `OptionalDuplicationSpan`, we can't accurately know how many duplicates there are, just how many we see.

This is kind of akin to the old C compiler problem where you can say get C preprocessor variables
`cc -DFOO` where their source location is on `<command-line>`.

Except here because we're dealing with `Option<Span>` ↝ `Vec<Span>`, these `None` values end up dropped on the floor.

Reply:
Anyhow it seems likely to me that these should be something else besides `Option<Span>`, perhaps something to the effect of:

```
enum Location {
   Span(Span),
   CommandLine,
   Other(String),
}
```

Allowing `Other("From CTParserBuilder".to_string())` or some such.

Reply:
Attempted a fix in [714d9da](https://github.com/softdevteam/grmtools/pull/547/commits/714d9dab9ec5e64320499de5e2d98ac409b21297) anyways